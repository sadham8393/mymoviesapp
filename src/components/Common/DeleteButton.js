import React from 'react';
import PropTypes from 'prop-types';

const DeleteButton = (props) =>{
    return (
        <button className = "waves-effect waves-light btn-small deleteBtn" onClick = {() => props.onDeleteClick(props.movie)}> Delete </button>
    )
}

DeleteButton.propTypes = { 
    movie: PropTypes.object, 
    onDeleteClick: PropTypes.func
} 

export default DeleteButton;