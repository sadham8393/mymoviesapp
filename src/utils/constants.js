// URl constants

export const MOVIEDB_PATH = "https://api.themoviedb.org/3/";
export const REACT_MOVIEDB_API_KEY  = "?api_key=0e89680c8b3fa0f259b53bbcc8f6091d";
export const MOVIEDB_IMAGE_APPEND_URL = "https://image.tmdb.org/t/p/w500/";
export const TRENDING_DAY_URL = "trending/all/day";


export const NO_IMAGE_URl = "https://s3-ap-southeast-1.amazonaws.com/upcode/static/default-image.jpg";

//Common Constants
export const INPUT_TYPE_TEXT = "text";
export const INPUT_TYPE_SELECT = "select";
export const NOT_AVAILABLE = "Not Available";


//Action Constants
export const FETCH_TRENDING_MOVIES = "FETCH_TRENDING_MOVIES";
export const FETCH_TV_MOVIES = "FETCH_TV_MOVIES";
export const DELETE_TV_MOVIES = "DELETE_TV_MOVIES";
export const MOVIES_FETCHED_SUCCESS = "MOVIES_FETCHED_SUCCESS";
export const MOVIES_FETCHED_ERROR = "MOVIES_FETCHED_ERROR";
export const DELETE_MOVIE_SUCCESS = "DELETE_MOVIE_SUCCESS";
export const DELETE_MOVIE_FAILURE = "DELETE_MOVIE_FAILURE";

export const MOVIES_FETCHED_NO_RESULTS = "MOVIES_FETCHED_NO_RESULTS";

export const RESET_RESPONSE = "RESET_RESPONSE";

export const NO_RESULTS_FOUND = "No results found for the given search input";

export const SERVER_UNAVAILABE = "Server unavailable. Please try after sometime"



