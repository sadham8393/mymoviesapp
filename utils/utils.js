import configureStore from 'redux-mock-store'

export const findByTestClass = (component, className) => {
    const wrapper = component.find(`${className}`);
    return wrapper;
};

export const findByComponent = (component, innerComponent) => {
    const wrapper = component.find(innerComponent);
    return wrapper;
};

export const testStore = (initialState) => {
    let store;
    const mockStore = configureStore();
    store = mockStore(initialState)
    return store;
};