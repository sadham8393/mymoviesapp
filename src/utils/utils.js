import { NOT_AVAILABLE } from './constants';

/**
 * Convert date format from "yyyy-mm-dd" to "dd/mm/yy"
 */

export function convertDate(date){
    if(date){
        return date.toString().split("-").reverse().join("/");
    }
    return NOT_AVAILABLE;
}