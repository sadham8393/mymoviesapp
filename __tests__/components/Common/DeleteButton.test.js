import React from "react";
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { findByTestClass } from '../../../utils/utils';
import DeleteButton  from "../../../src/components/Common/DeleteButton";

describe("Delete Button component", () => {
    describe('Renders correctly', () => {
        let wrapper;
        let mockFunc;
        let props;
        beforeEach(() => {
            mockFunc = jest.fn();
            props = {
                movie:{ title : "Avengers"} ,
                onDeleteClick : mockFunc
            };
            wrapper = shallow(<DeleteButton {...props} />);
        });

        it('should render correctly', () => {
            const props = {
                movie:{ title : "Avengers"},
                onDeleteClick : mockFunc
            };
            const tree = renderer.create(<DeleteButton {...props} />);
            expect(tree.toJSON()).toMatchSnapshot();
        });

        it('Should Render a button length to be 1', () => {
            const button = findByTestClass(wrapper, '.deleteBtn');
            expect(button.length).toBe(1);
        });

        it('Should emit callback on click event', () => {
            const button = findByTestClass(wrapper, '.deleteBtn');
            button.simulate('click');
            const callback = mockFunc.mock.calls.length;
            expect(callback).toBe(1);
        });
    })
});