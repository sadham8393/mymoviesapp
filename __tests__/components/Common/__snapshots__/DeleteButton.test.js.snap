// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Delete Button component Renders correctly should render correctly 1`] = `
<button
  className="waves-effect waves-light btn-small deleteBtn"
  onClick={[Function]}
>
   Delete 
</button>
`;
